src/Compiler/Service/service.fs

```diff
@@ -1514,17 +1527,19 @@ type FSharpChecker
             options: FSharpProjectOptions,
             ?userOpName: string
         ) =
-        let userOpName = defaultArg userOpName "Unknown"
-
-        backgroundCompiler.CheckFileInProjectAllowingStaleCachedResults(
-            parseResults,
-            fileName,
-            fileVersion,
-            SourceText.ofString source,
-            options,
-            userOpName
-        )
-        |> Async.AwaitNodeCode
+        async {
```

Comment:
interesting - you had to move to explicit async CE usage to ensure correct scoping of the resource lifetimes?

Reply:
Same as above - needed it before I added support for `.Using()`.